enhancements/machine-config/on-cluster-layering.md

```diff
@@ -0,0 +1,1137 @@
+| `.spec.renderedMachineConfig`     | Required                  | The name of the rendered MachineConfig built into this image (if one is built in). |
+| `.spec.machineConfigPool`         | Required                  | The name of the MachineConfigPool this MachineOSImage was built for.               |
+
+If a cluster-admin were so inclined, they could potentially create their own
+MachineOSImage from an image built off-cluster. This could eventually replace
+the `osImageURL` mechanism provided today by MachineConfigs.
+
+#### Nodes (corev1.Node)
+
+The following annotations will be added to each node object to indicate
+the current and desired image states:
+
+| **Annotation Name:**                             | **Description:**                                                                                |
+| ------------------------------------------------ | ----------------------------------------------------------------------------------------------- |
+| `machineconfiguration.openshift.io/currentImage` | The fully-qualified digested image pullspec for the currently-booted OS image for a given node. |
+| `machineconfiguration.openshift.io/desiredImage` | The fully-qualified digested image pullspec for the desired OS image for a given node.          |
+
+**Discussion:** It was previously discussed that using the `osImageURL`
+field on the MachineConfigs will be a more consistent way of doing this in
+the future, so this practice will likely go away.
+
+#### MachineConfigPools
+
+The MachineConfigPool CRD will be extended thusly:
+
+| **Field Name / Path (JSONPath):** | **Optional or Required:** | **Intended Use:**                                                                                                                                                                                                |
+| --------------------------------- | ------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `.spec.machineOSImage`            | Required                  | The name of the MachineOSImage that the pool should be running. Changing this will roll out a new MachineOSImage to all of the nodes in the MachineConfigPool.                                                   |
+| `.spec.deployMode`                | Optional                  | Controls how the MachineConfigs should be combined with the custom OS contents. Possible options are: `image` and `bootc`. See Implementation Details for details. (Not sure what a sensible default should be). |
+| `.status.osImagePullspec`         | Required                  | The OS image pullspec that is rolled out to all of the nodes.                                                                                                                                                    |
+
+#### ConfigMaps (corev1.ConfigMap)
```

Comment:
Configmap APIs do not age well, and we have seen many attempts in the past to move from configmaps to structured APIs

Why do you need a configmap, rather than creating a structured API to start with?